packages/syft/src/syft/client/client.py

```diff
@@ -646,19 +859,35 @@ def guest(self) -> Self:
             metadata=self.metadata,
         )
 
-    def exchange_route(self, client: Self) -> SyftSuccess | SyftError:
-        # relative
-        from ..service.network.routes import connection_to_route
-
-        self_node_route = connection_to_route(self.connection)
-        remote_node_route = connection_to_route(client.connection)
-        if client.metadata is None:
-            return SyftError(f"client {client}'s metadata is None!")
-        result = self.api.services.network.exchange_credentials_with(
-            self_node_route=self_node_route,
-            remote_node_route=remote_node_route,
-            remote_node_verify_key=client.metadata.to(NodeMetadataV3).verify_key,
-        )
+    def exchange_route(
+        self, client: Self, protocol: SyftProtocol = SyftProtocol.HTTP
+    ) -> SyftSuccess | SyftError:
+        if protocol == SyftProtocol.HTTP:
+            # relative
+            from ..service.network.routes import connection_to_route
+
+            self_node_route = connection_to_route(self.connection)
+            remote_node_route = connection_to_route(client.connection)
+            if client.metadata is None:
+                return SyftError(f"client {client}'s metadata is None!")
+
+            result = self.api.services.network.exchange_credentials_with(
+                self_node_route=self_node_route,
+                remote_node_route=remote_node_route,
+                remote_node_verify_key=client.metadata.to(NodeMetadataV3).verify_key,
+            )
+
+        elif protocol == SyftProtocol.VEILID:
+            # relative
+            from ..service.network.routes import connection_to_route
```

Comment:
Nitpick: we can move this import outside of the if else to the top

Reply:
Resolved by: [3c16d60](https://github.com/OpenMined/PySyft/pull/8526/commits/3c16d60e3956d1a4dacc54fcdee566ef3a18ab08)